### PR TITLE
added flag enum support

### PR DESCRIPTION
**Pull request by chrisspre**
- added syntactical construct for e.g. `flags Color { red green blue }`
- added transformation to CSDL enum with IsFlags=true

### CODE REVIEW

File: tools/rsdl/dotnet/rapid.rdm.transformation/ModelBuilder.cs [hunk 1]
```diff
@@ -77,12 +77,15 @@ private void AddSchemaElements(IRdmSchemaElement item)
...
-            var edmType = new EdmEnumType(rdmModel.Namespace.NamespaceName, definition.Name);
+            var edmType = new EdmEnumType(rdmModel.Namespace.NamespaceName, definition.Name, definition.IsFlags);
             edmModel.AddElement(edmType);
+            var flags = definition.IsFlags;
             for (int i = 0; i < definition.Members.Count; i++)
             {
                 var elem = definition.Members[i];
-                edmType.AddMember(new EdmEnumMember(edmType, elem, new EdmEnumMemberValue(i)));
+                var value = flags ? (1 << i) : i;
+                System.Console.WriteLine(value);
```
Comment (ralfhandl) [inline tools/rsdl/dotnet/rapid.rdm.transformation/ModelBuilder.cs:87]: Do you want to leave this in?

File: .vscode/settings.json [hunk 2]
```diff
@@ -1,11 +1,17 @@
...
+    "rsdl",
+    "tools",
+    "abnf",
+    "eset",
+    "jetsons",
+    "odata",
+    "svel",
+    "xunit"
+  ],
+  "dotnet-test-explorer.testProjectPath": "**/*tests.@(csproj|vbproj|fsproj)"
```
Comment (ralfhandl) [inline .vscode/settings.json:16]: Do you want to add cSpell and the dotnet-test-explorer to the list of recommended VS Code extensions in file `.vscode/extensions.json`?

File: tools/rsdl/dotnet/README.md [hunk 3]
```diff
@@ -9,10 +9,10 @@ The rapid cli program can be run from this folder with
...
 ```sh
 cd rapid
-dotnet run -- sample.rsdl --format JSON
-dotnet run -- sample.rsdl --format XML
-dotnet run -- sample.rsdl --format XML.JSON
-dotnet run -- sample.rsdl
+dotnet run -- samples/sample.rsdl --format JSON
+dotnet run -- samples/sample.rsdl --format XML
+dotnet run -- samples/sample.rsdl --format XML.JSON
+dotnet run -- samples/sample.rsdl
```
Comment (ralfhandl) [inline tools/rsdl/dotnet/README.md:15]: @chrisspre These commands currently fail with

```
Unhandled exception. System.InvalidCastException: Unable to cast object of type 'Microsoft.OData.Edm.EdmNavigationProperty' to type 'Microsoft.OData.Edm.IEdmStructuralProperty'.
   at rapid.rsdl.ModelBuilder.<>c__DisplayClass8_0.<AddStructuredType>b__0(RdmProperty key) in C:\SAPDevelop\odata-rapid\tools\rsdl\dotnet\rapid.rdm.transformation\ModelBuilder.cs:line 117
```

File: tools/rsdl/dotnet/rapid.rdm.transformation/ModelBuilder.cs (hunk 1 above)
Comment (chrisspre) [reply tools/rsdl/dotnet/rapid.rdm.transformation/ModelBuilder.cs]: Hello no. Thanks for catching this.

File: .vscode/settings.json (hunk 2 above)
Comment (chrisspre) [reply .vscode/settings.json:16]: Done in latest commit

File: tools/rsdl/dotnet/README.md (hunk 3 above)
Comment (chrisspre) [reply tools/rsdl/dotnet/README.md:15]: This turns out to be a fundamental bug in the way the transformation resolves type references.
Started to investigate this in PR #166 
Is not related to this PR and I would prefer to handle it separately.

**Pull request merged**